docs/spec.md

```diff
@@ -335,6 +335,10 @@ A `number` range includes all values of any numeric type (i.e. `float`, `int`, a
 When testing a value for inclusion, the range bounds and the value that is being tested will all be converted to
 `decimal` values for the sake of comparison. `number` ranges do not include values of any other type.
 
+The `float` type includes [special non-number values](https://amzn.github.io/ion-docs/docs/float.html#special-values) (`nan`, `+inf`, and `-inf`).
+The bounds of a range may not be any of these values, and attempting to use a non-number value as the bound of a number range will result in an error. 
```

Comment:
Consider dropping the 'may not' clause since that phrase is a bit gray when used in spec language.
```suggestion
Attempting to use a non-number value as the bound of a number range will result in an error. 
```